lib/postprocessing/image.js

```diff
@@ -178,6 +181,35 @@ async function needsImagePostProcess(rendition) {
     return false;
 }
 
+async function processWatermark(img, renditionInstructions, outdir) {
+    console.log('Processing watermark...');
+
+    const watermarkProcessor = new WatermarkProcessor(renditionInstructions);
+    const resizedWatermarkPath = renditionInstructions.watermark.name ? path.join(outdir, 'resized-'+ renditionInstructions.watermark.name)
+        : path.join(outdir, 'resized-'+ path.basename(renditionInstructions.watermark.path));
+
+    try {
+        await watermarkProcessor.fitWatermark(resizedWatermarkPath);
+    } catch(err) {
+        console.error('Error while resizing watermark to rendition dimensions', err);
+        // throw err; or should we revert to using original watermark as is i.e use renditionInstructions.watermark.path ?
+    }
+    
+    // 2. resizedWatermarkPath has new watermark to be applied onto infile
+
+    const placement = renditionInstructions.watermark.placement || 'Center';
+    // composite with crop, orient not supported
+    // https://github.com/aheckmann/gm/issues/549
+    // img.gravity(placement);
+    // img.composite(resizedWatermarkPath);
+
+    img.out(resizedWatermarkPath);
+    img.gravity(placement);
```

Comment:
I am not sure we want to use imagemagick's [gravity](https://www.imagemagick.org/script/command-line-options.php#gravity) 1:1 in our `watermark.placement` instruction.

Reply:
What alternative would you suggest? Do you have an idea in mind?